reflex/utils/prerequisites.py

```diff
@@ -665,18 +682,16 @@ def install_bun():
         console.debug("Skipping bun installation on Windows.")
         return
 
-    # Skip if bun is already installed.
-    if os.path.exists(get_config().bun_path) and get_bun_version() == version.parse(
-        constants.Bun.VERSION
-    ):
-        console.debug("Skipping bun installation as it is already installed.")
-        return
-
     #  if unzip is installed
     unzip_path = path_ops.which("unzip")
     if unzip_path is None:
         raise FileNotFoundError("Reflex requires unzip to be installed.")
 
+    # Skip if bun is already installed.
+    if check_bun_version():
+        console.debug("Skipping bun installation as it is already installed.")
+        return
+
```

Comment:
i think we should skip the bun installation _before_ we fail for not having unzip installed.

for example if the user was able to install bun themselves and doesn't want/need `unzip`, then we should still allow them to run if the version is appropriate.